src/com/nutrons/framework/inputs/CommonController.java

```diff
@@ -0,0 +1,148 @@
+package com.nutrons.framework.inputs;
+
+import io.reactivex.Flowable;
+
+public class CommonController extends WpiGamepad {
+  private final int axis1X;
+  private final int axis2X;
+  private final int axis1Y;
+  private final int axis2Y;
+  private final int ltrigger;
+  private final int rtrigger;
+  private int xButton;
+  private int aButton;
+  private int bButton;
+  private int yButton;
+  private int startButton;
+  private int rightBumper;
+  private int leftBumper;
+  private int selectButton;
+
+  private CommonController(int port,
+                           int axis1X,
+                           int axis1Y,
+                           int axis2X,
+                           int axis2Y,
+                           int lTrigger,
+                           int rTrigger,
+                           int buttonX,
+                           int buttonY,
+                           int buttonB,
+                           int buttonA,
+                           int buttonStart,
+                           int ButtonSelect,
+                           int bumperRight,
+                           int bumperleft) {
+    super(port);
+    this.axis1X = axis1X;
+    this.axis2X = axis2X;
+    this.axis1Y = axis1Y;
+    this.axis2Y = axis2Y;
+    this.ltrigger = lTrigger;
+    this.rtrigger = rTrigger;
+    this.xButton = buttonX;
+    this.yButton = buttonY;
+    this.bButton = buttonB;
+    this.aButton = buttonA;
+    this.startButton = buttonStart;
+    this.selectButton = ButtonSelect;
+    this.rightBumper = bumperRight;
+    this.leftBumper = bumperleft;
+
+
+  }
+
+  public static CommonController logitechF310(int port) {
```

Comment:
Not realistic, because it is a pain to handle if they try to make logitechF310 and xbox360 on the same port. leave like this for now.